advisories/a98779.md

```diff
@@ -0,0 +1,47 @@
+---
+title: Technical Advisory 98779
+advisory: A-98779
+summary: A restore job can potentially skip some data files upon resumption of an in-progress <code>RESTORE</code>, which could lead to missing rows after the job succeeds.
+toc: true
+affected_versions: v22.2.6
+advisory_date: 2023-03-29
+docs_area: releases
+---
+
+Publication date: {{ page.advisory_date | date: "%B %e, %Y" }}
+
+## Description
+
+A bug was introduced in CockroachDB v22.2.6 that can cause a restore job to skip some data files upon resumption of an in-progress [`RESTORE`](../{{site.versions["stable"]}}/restore.html). This could potentially lead to some missing rows after the restore job succeeds.
+
+A resumption could occur due to manual user action, such as [pausing](../{{site.versions["stable"]}}/pause-job.html)  and [resuming](../{{site.versions["stable"]}}/resume-job.html) the job, or, if a job is automatically retried internally, such as after a node restart. Therefore, this failure might not be noticed by an operator.
+
+## Statement
+
+This is resolved in CockroachDB by [#99066](https://github.com/cockroachdb/cockroach/pull/99066), which removes the incorrect optimization, and reverts back to the behavior of v22.2.5.
+
+The fix has been applied to maintenance releases of CockroachDB [v22.2.7](../releases/v22.2.html#v22-2-7) and later. This problem can also be mitigated immediately by updating the `SET CLUSTER SETTING bulkio.restore.use_simple_import_spans` cluster setting. See the following [Mitigation](#mitigation) section for details.
+
+This public issue is tracked by [#98779](https://github.com/cockroachdb/cockroach/issues/98779).
+
+## Mitigation
+
+Users of CockroachDB v22.2.6 are encouraged to upgrade to [v22.2.7](../releases/v22.2.html#v22-2-7) or a later version.
+
+Until such an upgrade, users of v22.2.6 can mitigate immediately by running:
+
+{% include_cached copy-clipboard.html %}
+```sql
+SET CLUSTER SETTING bulkio.restore.use_simple_import_spans = true;
+```
+
+This setting has already been automatically applied to all affected {{ site.data.products.db }} clusters.
+
+## Impact
+
+For users of CockroachDB v22.2.6, a restore that is resumed for any reason may give incorrect results. The resume could be due to manual job control actions, automatic retries if nodes restart, or after some types of internal rebalancing events.
+
+Enabling the cluster setting described in the [Mitigation](#mitigation) section, as well as upgrading to a later or earlier patch version, will prevent this.
```

Comment:
nit: `as well as` --> `or`

(two options, only one of which need be pursued)

nit: `or earlier` seems out of place. Do we support downgrades? Changing the cluster setting seems way less intrusive than a cluster downgrade to get around this issue. Recommend (unless engineering really pushed for this):

```
... upgrading to v22.2.6 or later, will prevent this issue.
```

Reply:
Yes, checked in with eng, we're remove downgrade. Thanks for pointing this out @andf-crl !